partiql-conformance-tests/build.rs

```diff
@@ -0,0 +1,296 @@
+fn to_full_file_name(prefix: &str, ion_file: &Path) -> String {
+    format!(
+        "{}_{}",
+        prefix,
+        ion_file
+            .with_extension("rs")
+            .file_name()
+            .unwrap()
+            .to_str()
+            .unwrap()
+    )
+}
+
+/// Converts the scope to a String that is passed through codegen's default formatter
+fn scope_to_formatted_string(scope: Scope) -> String {
+    let mut dest_str = String::new();
+    scope
+        .fmt(&mut Formatter::new(&mut dest_str))
+        .unwrap_or_else(|error| panic!("Failure during scope formatting: {:?}", error));
+    dest_str
+}
+
+trait TestFunction {
+    fn to_test_function(&self) -> Function;
+}
+
+/// Represents a conformance test with a PartiQL statement that will parse without errors
+/// TODO: add additional parse checks (e.g. ast)
+struct ParsePassTestCase {
+    test_name: String,
+    statement: String,
+    namespace: Option<String>,
+}
+
+impl TestFunction for ParsePassTestCase {
+    /// Creates a [`codegen::Function`] that checks the parsing of `statement` does not result in
+    /// a parsing error.
+    fn to_test_function(&self) -> Function {
+        let complete_test_name = match &self.namespace {
+            Some(ns) => format!("{}_{}", ns, &self.test_name),
+            None => self.test_name.to_owned(),
+        };
+        let mut generated_fun = Function::new(complete_test_name.replace(' ', "_").as_str());
+        generated_fun
+            .attr("test")
+            .line(format!(
+                "let parse_result = lalr_parse(\"{}\");",
+                &self.statement
+            ))
+            .line("assert!(parse_result.is_ok());");
+        generated_fun
+    }
+}
+
+/// Represents a conformance test with a PartiQL statement that will give an error when parsed
+/// TODO: add additional fields for further error-checking (e.g. error code)
+struct ParseFailTestCase {
+    test_name: String,
+    statement: String,
+    namespace: Option<String>,
+}
+
+impl TestFunction for ParseFailTestCase {
+    /// Creates a [`codegen::Function`] that checks the parsing of `statement` results in a parsing
+    /// error.
+    fn to_test_function(&self) -> Function {
+        let complete_test_name = match &self.namespace {
+            Some(ns) => format!("{}_{}", ns, &self.test_name),
+            None => self.test_name.to_owned(),
+        };
+        let mut generated_fun = Function::new(complete_test_name.replace(' ', "_").as_str());
+        generated_fun
+            .attr("test")
+            .line(format!(
+                "let parse_result = lalr_parse(\"{}\");",
+                &self.statement
+            ))
+            .line("assert!(parse_result.is_err());");
+        generated_fun
+    }
+}
+
+/// Parses the given .ion file `path` and creates a new [`codegen::Scope`] with a
+/// [`codegen::Function`] for each test in `path`.
+///
+/// Assumes the given tests in `path` all parse without errors.
```

Comment:
Currently, the data file parsing assumes the files have the correct schema and are at the correct directory (e.g. successful parse tests are in `pass/parse` directory). Can we make these assumptions, assuming there's data file validation done by `partiql-tests`? Otherwise, I can add more checks in the test runner, verifying the schema.

Reply:
I think there's likely utility in not making those assumptions blindly. 

For example, it might be nice to prototype a test case just by adding some test cases to an ion file and pointing this generator at it.

Reply:
I see these two (validation in partiql-tests and in each language runner) being somewhat independent as we might have different levels of validations based on requirements in target languages. Agree with Josh.